### PR TITLE
More consistent download experience for overwrite

### PR DESCRIPTION
**Pull request by rwilson504**
Power Platform Connectors CLI Update

This pull request will allow for a more consistent download experience and allow for better ALM by ensuring that custom connectors can be downloaded even if the files already exist.  Also updates the documentation which had no mention of the overwrite argument.

Issue: User cannot download connector files with the --overwrite argument unless they also supply the --destination argument. If they attempt to download the connector without the --destination argument the following error always appears, "{connectorid} directory already exists. Please remove the directory before continuing."

New Experience
* paconn download; If directory is found user is provided option to overwrite files within connectionid directory.
* paconn download --overwrite;  If directory is found files are automatically overwritten
* paconn download --destination c:\code\dest; If directory is found user is provided option to overwrite files within destination directory.
* paconn download --destination c:\code\dest --overwrite; If directory is found files are automatically overwritten

### CODE REVIEW

File: tools/paconn-cli/paconn/operations/download.py [hunk 1]
```diff
@@ -85,7 +84,7 @@ def download(powerapps_rp, settings, destination, overwrite):
     """
     # Prepare folders
     directory = _prepare_directory(
-        destination=destination,
+        destination=destination,
```
Comment (mamurshe) [inline tools/paconn-cli/paconn/operations/download.py:87]: Please remove the spaces here.

File: tools/paconn-cli/paconn/operations/download.py [hunk 2]
```diff
@@ -47,9 +47,8 @@ def _prepare_directory(destination, connector_id):
     # Create a sub-directory in the current directory
     # when a destination isn't provided
     else:
-        if os.path.isdir(connector_id):
-            error = '{} directory already exists. Please remove the directory before continuing.'
-            raise CLIError(error.format(connector_id))
+        if os.path.isdir(connector_id):
```
Comment (mamurshe) [inline tools/paconn-cli/paconn/operations/download.py:50]: Please simplify the logic a bit more like following:

```python
        if not os.path.isdir(connector_id):
            os.mkdir(connector_id)
        destination = connector_id
```

**Review by mamurshe — CHANGES_REQUESTED** [review]

File: tools/paconn-cli/paconn/operations/download.py (hunk 2 above)
Comment (rwilson504) [reply tools/paconn-cli/paconn/operations/download.py]: Thanks for the feedback!

**Comment by sriyen-msft** [comment]
@mamurshe Can you please review and approve PR

**Pull request merged**